### DISCUSSION TITLE
Xray(M+F+H+G+B+A)+Nginx/1_nginx.conf，这个请教一下大佬

### DISCUSSION THREAD

**Discussion by blog2020** [lxhao61/integrated-examples#213] [Q&A] [answered]
Xray(M+F+G+B+A)+Nginx/1_nginx.conf
请教一下大佬：
1、里面有段话“#若上面证书与下面证书由同一个通配符证书或 SAN 证书提供，此部分配置可删除、合并到下面配置中。注意：同步调整 Xray 示例中转发给自己的网站监听端口为如下监听端口。”
这段话的意思是这样吗：如果我的域名abc.com，申请了泛域名证书，是不是说( listen 127.0.0.1:2443)这一段的server块可以删除，只用下面7443的server块，并把（ssl_certificate /home/tls/zh.xx.yy/zh.xx.yy.crt;）改成：（ssl_certificate /home/tls/abc.com/abc.com.crt;）
2、( location /SALdGZ9k { #与 Shadowsocks+gRPC 应用中 serviceName 对应)，这个location出现了两次，不可以合并吗？

谢谢。

**Comment by lxhao61** [comment, marked as answer]
1、是的。另外注意：同步调整 Xray 示例中转发给自己的网站监听端口为如下监听端口。
2、不可以合并。见备注，分别对应一个域名应用。另外可以把此重复部分提取出来单独成一个配置文件，使用插入配置分别引用。

**Reply by blog2020** [reply]
大佬你好，自从按照你的配置搭建后一直稳定使用。现在有三个问题想请教一下大佬，为了不占用太多页面，我就在这贴写了：
1、我看到有些教程说最好加上ssl_reject_handshake，这个在Xray(M+F+G+B+A)+Nginx中需要添加吗？
2、我有台被墙IP的机器，想加上cdn用用，是用vless好还是vmess？听说vless没有加密，看了一下vmess好像没有alterID参数了（新版vmess吗）
3、Xray(VMess+HTTPUpgrade)+Nginx\Caddy这个的Nginx配置里启用了Tls1.3和x25519，如果套上cdn，会不会被人用reality偷流量（被当作cdn的一个节点）？ 谢谢。

**Reply by lxhao61** [reply]
1、无必要，见 Nginx 配置下对应备注。
2、见首页中对应注意。VMess 早就不需要 alterID 参数。
3、不知道你怎么实现用 REALITY 偷流量？

**Comment by lxhao61** [comment]
参考新的示例，重复如 Shadowsocks+gRPC+TLS 应用等配置己合并。
